src/main/java/core/basesyntax/ArrayList.java

```diff
@@ -1,48 +1,132 @@
 package core.basesyntax;
 
+import java.util.NoSuchElementException;
+
 public class ArrayList<T> implements List<T> {
+    private static final int DEFAULT_CAPACITY = 10;
+    private static final double RESIZE_COEFFICIENT = 1.5;
+    private Object[] array;
+    private Object[] tempArray;
+    private int size = 0;
+
+    public ArrayList(int capacity) {
+        if (capacity > 0) {
+            this.array = new Object[capacity];
+        } else {
+            this.array = new Object[DEFAULT_CAPACITY];
+        }
+    }
+
+    public ArrayList() {
+        this.array = new Object[DEFAULT_CAPACITY];
+    }
+
     @Override
     public void add(T value) {
-
+        if (this.size + 1 > array.length) {
+            resizeArray(array);
+        }
+        array[size] = value;
+        size++;
     }
 
     @Override
     public void add(T value, int index) {
+        if (index > size || index < 0) {
+            throw new ArrayListIndexOutOfBoundsException("Wrong index " + index);
+        } else if (index == size) {
+            this.add(value);
+        } else {
+            tempArray = new Object[size - index];
+            System.arraycopy(array, index, tempArray, 0, size - index);
```

Comment:
you dont need a temporary array - you can copy values to the same array

Reply:
I needed temporary array to call add() in cycle. I took into account your comment and rewrite this part.